scripts/gcov_process.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -e
+
+# -b is used for binary dump and -c is used for console log
+if [[ $# -ne 3 ]]; then
+	echo "Usage: $0 [-b <binary_dump> | -c <console_text>] <build_dir>"
+	exit
+fi
+
+if [ -n "${CROSS_COMPILE}" ]; then
+	GCOV=${CROSS_COMPILE}/gcov
+else
+	GCOV=$(which gcov)
+fi
+
+if [ -z "$GCOV" ]; then
+	echo "$0: Could not find gcov"
+	exit
+fi
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}"  )" &> /dev/null && pwd  )
```

Comment:
Somehow I missed this in the last iteration. What are the benefits of this compared to `$(dirname $0)`?

Reply:
$(dirname $0) returns a relative path to the file (opposed to my way in which I get the absolute path of the script) which becomes a problem when this `cd ${GCOV_DIR}/scripts` instruction executes. Maybe I can still adapt it to make it simpler, will look into it.